tests/wycheproof/test_hmac.py

```diff
@@ -39,7 +40,10 @@ def test_hmac(backend, wycheproof):
     hash_algo = _HMAC_ALGORITHMS[wycheproof.testfiledata["algorithm"]]
     if wycheproof.testgroup["tagSize"] // 8 != hash_algo.digest_size:
         pytest.skip("Truncated HMAC not supported")
-    if not backend.hmac_supported(hash_algo):
+    if not backend.hmac_supported(hash_algo) or (
+        rust_openssl.CRYPTOGRAPHY_IS_AWSLC
+        and hash_algo.name.startswith("sha3-")
```

Comment:
Why doesn't `hmac_supported` handled this case correctly?

Reply:
Probably cipher_registry is overly heavy weight, but I do think we should push this logic into `hmac_supported`.